sql/parser/builtins.go

```diff
@@ -1161,6 +1186,31 @@ func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
 	return fmt.Sprintf("(?%s:%s)", flagString.String(), pattern), nil
 }
 
+func overlay(s, to string, pos, size int) (Datum, error) {
+	if pos < 1 {
+		return nil, errors.New("negative substring length not allowed")
+	}
+
+	var out bytes.Buffer
+	i := 0
+	for _, c := range s {
```

Comment:
Yeah, maybe that's true. Still, I think I'd do the `[]rune` thing in this change, and then make a follow-up change that introduces some common efficient string slicing function and use it everywhere. See https://github.com/cockroachdb/cockroach/pull/3076/files for instance.